Shopkeeper/Shopkeeper.Services/ItemService.cs

```diff
@@ -25,22 +28,33 @@ public ItemService(IItemRepository itemRepository)
 
         public Item GetRandom(params string[] exclude)
         {
+            if (exclude == null) exclude = Array.Empty<string>();
+
+            logger.LogInformation("Getting Random Item");
+            logger.LogDebug("Excluding: {0}", string.Join(',', exclude));
+
             var items = GetAll()
                 .Where(x => x.HasComponents && !exclude.Contains(x.Id));
 
             var index = Random.Next(items.Count());
 
+            logger.LogInformation("Random Item: {0}", items.ElementAt(index));
+
             return items.ElementAt(index);
         }
 
-        public IEnumerable<Item> GetFillerItems(int numberOfItems, params string[] exclude)
+        public IEnumerable<Item> GetFillerItems(int numberOfItems, IEnumerable<string> exclude)
         {
             var items = GetAll()
                 .Where(x => !exclude.Contains(x.Id));
```

Comment:
May want to materialize this to an array so you can use the indexer and .Count below